mis_builder/models/mis_report_instance.py

```diff
@@ -319,6 +322,12 @@ def _compute_dates(self):
         ),
     ]
 
+    @api.depends("report_instance_id")
+    def _compute_allowed_cmpcol_ids(self):
+        """Compute actual records while in NewId context"""
+        for record in self:
+            record.allowed_cmpcol_ids = record.report_instance_id.period_ids._origin
```

Comment:
`_origin` contains the values in `NewId` mode?

Reply:
At least that is what I understand by the documentation: https://github.com/odoo/odoo/blob/14.0/odoo/models.py#L5511
And other references: https://github.com/odoo/odoo/pull/49701#issue-405014807

In this case, `record.report_instance_id.period_ids` will have a set of NewId records like
```python
mis.report.instance.period(<NewId origin=2>, <NewId origin=3>, <NewId 0x7fdd275c5820>)
```
where the first 2 refer to existing columns, and the 3rd to the new one being created.
When we access `_origin`, we get:
```python
mis.report.instance.period(2, 3)
```
which points only to the existing records, skipping the one that is being created.

Reply:
And why skip them? You can do an assignation of `record.report_instance_id.period_ids` (removing itself), and it should work also for virtual ones.

Reply:
Do you mean what I do now in the compute? https://github.com/OCA/mis-builder/pull/361/files#diff-0f911a0342a8e73e0643be069c2eef52c082d983131fc6fa055428b49e17b353R330
Or do you mean doing this directly on the field's domain as before?

Reply:
I mean that the compute to be:
```python
record.allowed_cmpcol_ids = record.report_instance_id.period_ids - record
```

Reply:
Thanks, you are correct :+1: This works, so I just left it as you suggested